src/libs/actions/Report.ts

```diff
@@ -175,10 +176,27 @@ const typingWatchTimers: Record<string, NodeJS.Timeout> = {};
 
 let reportIDDeeplinkedFromOldDot: string | undefined;
 Linking.getInitialURL().then((url) => {
-    const params = new URLSearchParams(url ?? '');
-    const exitToRoute = params.get('exitTo') ?? '';
-    const {reportID} = ReportUtils.parseReportRouteParams(exitToRoute);
-    reportIDDeeplinkedFromOldDot = reportID;
+    const isWeb = ([CONST.PLATFORM.WEB] as unknown as string).includes(getPlatform());
+    const currentParams = new URLSearchParams(url ?? '');
+    const currentExitToRoute = currentParams.get('exitTo') ?? '';
+    const {reportID: currentReportID} = ReportUtils.parseReportRouteParams(currentExitToRoute);
+
+    if (!isWeb) {
+        reportIDDeeplinkedFromOldDot = currentReportID;
```

Comment:
isn't it possible that someone could click a link in OldDot web and have it open a native app?

Reply:
I tested this scenario a few times. It always opens a browser to handle a deep link. It proposes to open a native app, once you get a report screen opened. Therefore, it does not need to handle the deep link anymore.